code/iomethods.py

```diff
@@ -250,108 +295,82 @@ def read_data(varName, latName, lonName, userVariables, filelist=None):
-        userVariables.filelist = glob.glob(filelistInstructions)
-        
-    userVariables.filelist.sort()
+    tmp = netCDF4.Dataset(userVariables.filelist[0], 'r+', format='NETCDF4')
 
-    inputData = []
-    timelist = []
-    time2store = None
-    tempMaskedValueNp = []
+    alllatsraw = tmp.variables[latName][:]
+    alllonsraw = tmp.variables[lonName][:]
+    alllonsraw[alllonsraw > 180] = alllonsraw[alllonsraw > 180] - 360.  # convert to -180,180 if necessary
 
-    nfiles = len(userVariables.filelist)
+    tmp.close()
 
-    # Crash nicely if there are no netCDF files
-    if nfiles == 0:
-        print 'Error: no files in this directory! Exiting elegantly'
-        sys.exit()
-    else:
-        # Open the first file in the list to read in lats, lons and generate the  grid for comparison
-        tmp = Dataset(userVariables.filelist[0], 'r+', format='NETCDF4')
-
-        alllatsraw = tmp.variables[latName][:]
-        alllonsraw = tmp.variables[lonName][:]
-        alllonsraw[alllonsraw > 180] = alllonsraw[alllonsraw > 180] - 360.  # convert to -180,180 if necessary
-
-        # Get the lat/lon info data (different resolution)
-        latminNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMIN))
-        latmaxNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMAX))
-        lonminNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMIN))
-        lonmaxNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMAX))
-        latminIndex = (np.where(alllatsraw == latminNETCDF))[0][0]
-        latmaxIndex = (np.where(alllatsraw == latmaxNETCDF))[0][0]
-        lonminIndex = (np.where(alllonsraw == lonminNETCDF))[0][0]
-        lonmaxIndex = (np.where(alllonsraw == lonmaxNETCDF))[0][0]
-
-        # Subsetting the data
-        latsraw = alllatsraw[latminIndex: latmaxIndex]
-        lonsraw = alllonsraw[lonminIndex:lonmaxIndex]
-
-        LON, LAT = np.meshgrid(lonsraw, latsraw)
-
-        latsraw = []
-        lonsraw = []
-        tmp.close()
-
-    for files in userVariables.filelist:
+    # Get the lat/lon info data (different resolution)
+    latminNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMIN))
+    latmaxNETCDF = utils.find_nearest(alllatsraw, float(userVariables.LATMAX))
+    lonminNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMIN))
+    lonmaxNETCDF = utils.find_nearest(alllonsraw, float(userVariables.LONMAX))
+    latminIndex = (np.where(alllatsraw == latminNETCDF))[0][0]
+    latmaxIndex = (np.where(alllatsraw == latmaxNETCDF))[0][0]
+    lonminIndex = (np.where(alllonsraw == lonminNETCDF))[0][0]
+    lonmaxIndex = (np.where(alllonsraw == lonmaxNETCDF))[0][0]
+
+    # Subset the data
+    latsraw = alllatsraw[latminIndex:latmaxIndex]
+    lonsraw = alllonsraw[lonminIndex:lonmaxIndex]
+
+    LON, LAT = np.meshgrid(lonsraw, latsraw)
+
+    timeName = 'time'
+    for file in userVariables.filelist:
```

Comment:
file should be avoided as variable name. In general,  please be mindful of reserved words being used for variable or method names.